snapcraft/plugins/catkin.py

```diff
@@ -127,77 +160,59 @@ def gcc_version(self):
     @property
     def rosdir(self):
         return os.path.join(self.installdir,
-                            'opt', 'ros', self.options.rosversion)
+                            'opt', 'ros', self.options.rosdistro)
 
-    def _deps_from_packagesxml(self, f, pkg):
-        tree = lxml.etree.parse(f)
+    def _find_dependencies(self):
```

Comment:
I don't believe so, although not saving it as an attribute would mean I'd have to interrogate the sources for their dependencies in both `pull()` and `build()`. Depending on the size of the workspace that could be a bit heavy.